cross-compiling/building universal binaries on macOS broken?

What's the correct way to include libultrahdr in a universal (arm64 + x86_64) application on macOS?
 
I have successfully incorporated libultrahdr in my project http://github.com/wkjarosz/HDRView when I build purely for the native architecture of my M1 macbook. 

However, if I try to build a universal binary by passing in the following options during configure:

```
cmake -B build/test -G Ninja -DCMAKE_OSX_DEPLOYMENT_TARGET=11 -DCMAKE_OSX_ARCHITECTURES="arm64;x86_64" -DHELLOIMGUI_USE_FREETYPE=0
```

I get errors that seem to be due to libultrahdr setting an invalid CPU type:

```
[76/390] Building CXX object _deps/libultrahdr-build/CMakeFiles/im...r/third_party/image_io/src/base/byte_pointer_data_destination.cc.o
FAILED: _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_pointer_data_destination.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/includes -I/Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/src/modp_b64 -I/Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/src/modp_b64/modp_b64 -stdlib=libc++ -D_LIBCPP_VERSION -fvisibility=hidden -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_pointer_data_destination.cc.o -MF _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_pointer_data_destination.cc.o.d -o _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_pointer_data_destination.cc.o -c /Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/src/base/byte_pointer_data_destination.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
[79/390] Building CXX object _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_buffer.cc.o
FAILED: _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_buffer.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/includes -I/Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/src/modp_b64 -I/Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/src/modp_b64/modp_b64 -stdlib=libc++ -D_LIBCPP_VERSION -fvisibility=hidden -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_buffer.cc.o -MF _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_buffer.cc.o.d -o _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_buffer.cc.o -c /Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/src/base/byte_buffer.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
[81/390] Building CXX object _deps/libultrahdr-build/CMakeFiles/im...third_party/image_io/src/base/data_range_tracking_destination.cc.o
FAILED: _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/data_range_tracking_destination.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/includes -I/Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/src/modp_b64 -I/Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/src/modp_b64/modp_b64 -stdlib=libc++ -D_LIBCPP_VERSION -fvisibility=hidden -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/data_range_tracking_destination.cc.o -MF _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/data_range_tracking_destination.cc.o.d -o _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/data_range_tracking_destination.cc.o -c /Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/src/base/data_range_tracking_destination.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
[82/390] Building CXX object _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/data_line_map.cc.o
FAILED: _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/data_line_map.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/includes -I/Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/src/modp_b64 -I/Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/src/modp_b64/modp_b64 -stdlib=libc++ -D_LIBCPP_VERSION -fvisibility=hidden -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/data_line_map.cc.o -MF _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/data_line_map.cc.o.d -o _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/data_line_map.cc.o -c /Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/src/base/data_line_map.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
[83/390] Building CXX object _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/data_context.cc.o
FAILED: _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/data_context.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/includes -I/Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/src/modp_b64 -I/Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/src/modp_b64/modp_b64 -stdlib=libc++ -D_LIBCPP_VERSION -fvisibility=hidden -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/data_context.cc.o -MF _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/data_context.cc.o.d -o _deps/libultrahdr-build/CMakeFiles/image_io.dir/third_party/image_io/src/base/data_context.cc.o -c /Users/wjarosz/.cache/CPM/libultrahdr/5dadc195c68e380bfc09d7233f3f573a9cc78ed8/third_party/image_io/src/base/data_context.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
[87/390] Building CXX object _deps/openexr-build/src/lib/Iex/CMakeFiles/Iex.dir/IexBaseExc.cpp.o
ninja: build stopped: subcommand failed.
```

As a workaround, I tried setting `UHDR_ENABLE_INTRINSICS=0`, but the unknown target CPU is still set via `-march` regardless of whether intrinsics are enabled.

Here is a minimal reproducing example trying simply to build libultradhdr and its example app:

```
% ccmake -B build/universal1 -G Ninja -DCMAKE_OSX_DEPLOYMENT_TARGET=11 -DCMAKE_OSX_ARCHITECTURES="arm64;x86_64"
CMake Deprecation Warning at CMakeLists.txt:22 (cmake_policy):
  The OLD behavior for policy CMP0091 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- The C compiler identification is AppleClang 16.0.0.16000026
-- The CXX compiler identification is AppleClang 16.0.0.16000026
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- No build type chosen, selecting Release
-- No target type chosen, selecting Shared
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found JPEG: /opt/homebrew/lib/libjpeg.dylib (found version "80")
-- Configuring done (1.0s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/wjarosz/Documents/Work/Sandbox/libultrahdr/build/universal1
wjarosz@X7D61N7PP7 libultrahdr % cmake --build build/universal1 --parallel                                                                   
[1/55] Building CXX object CMakeFiles/image_io.dir/third_party/image_io/src/base/data_segment_data_source.cc.o
FAILED: CMakeFiles/image_io.dir/third_party/image_io/src/base/data_segment_data_source.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/includes -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64 -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64/modp_b64 -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -DUHDR_BUILDING_SHARED_LIBRARY -DUHDR_ENABLE_INTRINSICS -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT CMakeFiles/image_io.dir/third_party/image_io/src/base/data_segment_data_source.cc.o -MF CMakeFiles/image_io.dir/third_party/image_io/src/base/data_segment_data_source.cc.o.d -o CMakeFiles/image_io.dir/third_party/image_io/src/base/data_segment_data_source.cc.o -c /Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/base/data_segment_data_source.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
[2/55] Building CXX object CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_pointer_data_destination.cc.o
FAILED: CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_pointer_data_destination.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/includes -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64 -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64/modp_b64 -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -DUHDR_BUILDING_SHARED_LIBRARY -DUHDR_ENABLE_INTRINSICS -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_pointer_data_destination.cc.o -MF CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_pointer_data_destination.cc.o.d -o CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_pointer_data_destination.cc.o -c /Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/base/byte_pointer_data_destination.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
[3/55] Building CXX object CMakeFiles/image_io.dir/third_party/image_io/src/base/data_segment.cc.o
FAILED: CMakeFiles/image_io.dir/third_party/image_io/src/base/data_segment.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/includes -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64 -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64/modp_b64 -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -DUHDR_BUILDING_SHARED_LIBRARY -DUHDR_ENABLE_INTRINSICS -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT CMakeFiles/image_io.dir/third_party/image_io/src/base/data_segment.cc.o -MF CMakeFiles/image_io.dir/third_party/image_io/src/base/data_segment.cc.o.d -o CMakeFiles/image_io.dir/third_party/image_io/src/base/data_segment.cc.o -c /Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/base/data_segment.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
[4/55] Building CXX object CMakeFiles/image_io.dir/third_party/image_io/src/base/data_range_tracking_destination.cc.o
FAILED: CMakeFiles/image_io.dir/third_party/image_io/src/base/data_range_tracking_destination.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/includes -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64 -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64/modp_b64 -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -DUHDR_BUILDING_SHARED_LIBRARY -DUHDR_ENABLE_INTRINSICS -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT CMakeFiles/image_io.dir/third_party/image_io/src/base/data_range_tracking_destination.cc.o -MF CMakeFiles/image_io.dir/third_party/image_io/src/base/data_range_tracking_destination.cc.o.d -o CMakeFiles/image_io.dir/third_party/image_io/src/base/data_range_tracking_destination.cc.o -c /Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/base/data_range_tracking_destination.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
[5/55] Building CXX object CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_buffer.cc.o
FAILED: CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_buffer.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/includes -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64 -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64/modp_b64 -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -DUHDR_BUILDING_SHARED_LIBRARY -DUHDR_ENABLE_INTRINSICS -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_buffer.cc.o -MF CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_buffer.cc.o.d -o CMakeFiles/image_io.dir/third_party/image_io/src/base/byte_buffer.cc.o -c /Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/base/byte_buffer.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
[6/55] Building CXX object CMakeFiles/image_io.dir/third_party/image_io/src/base/data_line_map.cc.o
FAILED: CMakeFiles/image_io.dir/third_party/image_io/src/base/data_line_map.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/includes -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64 -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64/modp_b64 -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -DUHDR_BUILDING_SHARED_LIBRARY -DUHDR_ENABLE_INTRINSICS -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT CMakeFiles/image_io.dir/third_party/image_io/src/base/data_line_map.cc.o -MF CMakeFiles/image_io.dir/third_party/image_io/src/base/data_line_map.cc.o.d -o CMakeFiles/image_io.dir/third_party/image_io/src/base/data_line_map.cc.o -c /Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/base/data_line_map.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
[7/55] Building CXX object CMakeFiles/image_io.dir/third_party/image_io/src/base/istream_ref_data_source.cc.o
FAILED: CMakeFiles/image_io.dir/third_party/image_io/src/base/istream_ref_data_source.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/includes -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64 -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64/modp_b64 -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -DUHDR_BUILDING_SHARED_LIBRARY -DUHDR_ENABLE_INTRINSICS -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT CMakeFiles/image_io.dir/third_party/image_io/src/base/istream_ref_data_source.cc.o -MF CMakeFiles/image_io.dir/third_party/image_io/src/base/istream_ref_data_source.cc.o.d -o CMakeFiles/image_io.dir/third_party/image_io/src/base/istream_ref_data_source.cc.o -c /Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/base/istream_ref_data_source.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
[8/55] Building CXX object CMakeFiles/image_io.dir/third_party/image_io/src/base/ostream_ref_data_destination.cc.o
FAILED: CMakeFiles/image_io.dir/third_party/image_io/src/base/ostream_ref_data_destination.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/includes -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64 -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64/modp_b64 -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -DUHDR_BUILDING_SHARED_LIBRARY -DUHDR_ENABLE_INTRINSICS -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT CMakeFiles/image_io.dir/third_party/image_io/src/base/ostream_ref_data_destination.cc.o -MF CMakeFiles/image_io.dir/third_party/image_io/src/base/ostream_ref_data_destination.cc.o.d -o CMakeFiles/image_io.dir/third_party/image_io/src/base/ostream_ref_data_destination.cc.o -c /Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/base/ostream_ref_data_destination.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
[9/55] Building CXX object CMakeFiles/image_io.dir/third_party/image_io/src/base/message_writer.cc.o
FAILED: CMakeFiles/image_io.dir/third_party/image_io/src/base/message_writer.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/includes -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64 -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64/modp_b64 -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -DUHDR_BUILDING_SHARED_LIBRARY -DUHDR_ENABLE_INTRINSICS -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT CMakeFiles/image_io.dir/third_party/image_io/src/base/message_writer.cc.o -MF CMakeFiles/image_io.dir/third_party/image_io/src/base/message_writer.cc.o.d -o CMakeFiles/image_io.dir/third_party/image_io/src/base/message_writer.cc.o -c /Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/base/message_writer.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
[10/55] Building CXX object CMakeFiles/image_io.dir/third_party/image_io/src/base/message_handler.cc.o
FAILED: CMakeFiles/image_io.dir/third_party/image_io/src/base/message_handler.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/includes -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64 -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64/modp_b64 -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -DUHDR_BUILDING_SHARED_LIBRARY -DUHDR_ENABLE_INTRINSICS -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT CMakeFiles/image_io.dir/third_party/image_io/src/base/message_handler.cc.o -MF CMakeFiles/image_io.dir/third_party/image_io/src/base/message_handler.cc.o.d -o CMakeFiles/image_io.dir/third_party/image_io/src/base/message_handler.cc.o -c /Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/base/message_handler.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
[11/55] Building CXX object CMakeFiles/image_io.dir/third_party/image_io/src/base/data_scanner.cc.o
FAILED: CMakeFiles/image_io.dir/third_party/image_io/src/base/data_scanner.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/includes -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64 -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64/modp_b64 -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -DUHDR_BUILDING_SHARED_LIBRARY -DUHDR_ENABLE_INTRINSICS -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT CMakeFiles/image_io.dir/third_party/image_io/src/base/data_scanner.cc.o -MF CMakeFiles/image_io.dir/third_party/image_io/src/base/data_scanner.cc.o.d -o CMakeFiles/image_io.dir/third_party/image_io/src/base/data_scanner.cc.o -c /Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/base/data_scanner.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
[12/55] Building CXX object CMakeFiles/image_io.dir/third_party/image_io/src/base/data_context.cc.o
FAILED: CMakeFiles/image_io.dir/third_party/image_io/src/base/data_context.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/includes -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64 -I/Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/modp_b64/modp_b64 -O3 -DNDEBUG -std=c++17 -arch arm64 -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk -mmacosx-version-min=11 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -DUHDR_BUILDING_SHARED_LIBRARY -DUHDR_ENABLE_INTRINSICS -DUHDR_WRITE_ISO -ffunction-sections -fdata-sections -fomit-frame-pointer -ffp-contract=fast -march=armv8-a -fno-lax-vector-conversions -MD -MT CMakeFiles/image_io.dir/third_party/image_io/src/base/data_context.cc.o -MF CMakeFiles/image_io.dir/third_party/image_io/src/base/data_context.cc.o.d -o CMakeFiles/image_io.dir/third_party/image_io/src/base/data_context.cc.o -c /Users/wjarosz/Documents/Work/Sandbox/libultrahdr/third_party/image_io/src/base/data_context.cc
error: unknown target CPU 'armv8-a'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, raptorlake, meteorlake, sierraforest, grandridge, graniterapids, graniterapids-d, emeraldrapids, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, znver4, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
ninja: build stopped: subcommand failed.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -777,6 +777,7 @@ endif()
 target_link_libraries(${UHDR_TARGET_NAME} PRIVATE ${PRIVATE_LINK_LIBS})
 set_target_properties(${UHDR_TARGET_NAME}
                       PROPERTIES PUBLIC_HEADER ultrahdr_api.h)
+target_include_directories(${UHDR_TARGET_NAME} PUBLIC ${EXPORT_INCLUDE_DIR})
 if(BUILD_SHARED_LIBS)
   # If target is STATIC no need to set VERSION and SOVERSION
   set_target_properties(${UHDR_TARGET_NAME}
@@ -795,6 +796,7 @@ if(BUILD_SHARED_LIBS)
     target_link_libraries(${UHDR_TARGET_NAME_STATIC} PRIVATE ${log-lib})
   endif()
   target_link_libraries(${UHDR_TARGET_NAME_STATIC} PRIVATE ${PRIVATE_LINK_LIBS})
+  target_include_directories(${UHDR_TARGET_NAME_STATIC} PUBLIC ${EXPORT_INCLUDE_DIR})
   combine_static_libs(${UHDR_CORE_LIB_NAME} ${UHDR_TARGET_NAME_STATIC})
   if(NOT MSVC)
     set_target_properties(${UHDR_TARGET_NAME_STATIC}

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -68,6 +68,26 @@ else()
   message(FATAL_ERROR "Architecture: ${CMAKE_SYSTEM_PROCESSOR} not recognized")
 endif()
 
+# Cross compiling on macOS. The cross compiling architecture should override
+# auto-detected system architecture settings.
+if(CMAKE_OSX_ARCHITECTURES)
+  if(APPLE)
+    if(CMAKE_OSX_ARCHITECTURES STREQUAL "arm64")
+      set(ARCH "aarch64")
+    elseif(CMAKE_OSX_ARCHITECTURES STREQUAL "x86_64")
+      set(ARCH "x86_64")
+    elseif(CMAKE_OSX_ARCHITECTURES STREQUAL "i386")
+      set(ARCH "x86")
+    else()
+      message(FATAL_ERROR "Architecture ${CMAKE_OSX_ARCHITECTURES} is not "
+                          "supported. Only one architecture (arm64, x86_64 "
+                          "or i386) could be specified at build time.")
+    endif()
+  else()
+    message(WARNING "ignoring setting CMAKE_OSX_ARCHITECTURES:= ${CMAKE_OSX_ARCHITECTURES}")
+  endif()
+endif()
+
 ###########################################################
 # Directories
 ###########################################################

